src/public/components/restricted-input.js

```diff
@@ -0,0 +1,43 @@
+import K, * as U from "karet.util"
+import React     from "karet"
+
+export const RestrictedInput = ({value, meta: {format, parse}, ...props}) => {
+  const edited = U.atom(undefined)
```

Comment:
I mean that the local `edited` variable would simply be exposed as a parameter.  This allows a user of the `RestrictedInput` to store the `edited` state outside of the component.  I would still recommend giving `edited` a default value (making it local state, unless the user decides otherwise), because in most cases that will work just fine.  IOW:

```js
export const RestrictedInput = ({value, meta: {format, parse}, edited = U.atom(), ...props}) => {
```